### PR TITLE
feat: Check missing dependencies for Monokle

### PR DESCRIPTION
**Pull request by erdkse**
This PR...

## Changes

* Checks missing dependencies for Monokle application.
* Shows a warning notification to notify users.

## Fixes

* 

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -29,6 +33,15 @@ const App = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  ipcRenderer.once('missing-dependecy-result', (_, {dependencies}) => {
+    const alert: AlertType = {
+      type: AlertEnum.Warning,
+      title: 'Missing dependecy',
+      message: `${dependencies.toString()} must be installed.`,
```
Comment (olensmar) [inline src/App.tsx:40]: "...for all Monokle functionality to be available"

File: electron/main.ts [hunk 2]
```diff
@@ -13,8 +13,11 @@ import terminal from '../cli/terminal';
 
 Object.assign(console, ElectronLog.functions);
 
+const {which} = require('shelljs');
 const ElectronStore = require('electron-store');
 
+const APP_DEPENDENCIES = ['kubectl', 'helm', 'erdkse'];
```
Comment (olensmar) [inline electron/main.ts:19]: erdkse ? :D

File: electron/main.ts [hunk 3]
```diff
@@ -144,6 +155,14 @@ const openApplication = async (givenPath?: string) => {
   ElectronStore.initRenderer();
   const win = createWindow();
 
+  const missingDependecies = APP_DEPENDENCIES.filter(d => !which(d));
+
+  if (missingDependecies.length > 0) {
+    win.webContents.on('did-finish-load', () => {
+      win.webContents.send('missing-dependecy-result', {dependencies: missingDependecies});
```
Comment (olensmar) [inline electron/main.ts:162]: typo in "missing-dependecy-result" - should be "missing-dependency-result" in all places...

File: electron/main.ts [hunk 4]
```diff
@@ -43,6 +46,14 @@ ipcMain.on('run-kustomize', (event, folder: string) => {
   }
 });
 
+ipcMain.on('check-missing-dependency', event => {
```
Comment (olensmar) [inline electron/main.ts:49]: this doesn't seem to be called from anywhere!?

File: electron/main.ts [hunk 5]
```diff
@@ -144,6 +155,14 @@ const openApplication = async (givenPath?: string) => {
   ElectronStore.initRenderer();
   const win = createWindow();
 
+  const missingDependecies = APP_DEPENDENCIES.filter(d => !which(d));
```
Comment (olensmar) [inline electron/main.ts:158]: move to common function that is called from here and above? - i.e. "checkMissingDependencies"

File: electron/main.ts (hunk 2 above)
Comment (erdkse) [reply electron/main.ts:19]: To test purposes :D I will delete it

File: src/redux/thunks/utils.ts [hunk 6]
```diff
@@ -48,6 +49,7 @@ export function createPreviewResult(resourcesYaml: string, previewResourceId: st
  */
 
 export function createPreviewRejection(thunkAPI: any, title: string, message: string) {
+  ipcRenderer.send('check-missing-dependency');
```
Comment (erdkse) [inline src/redux/thunks/utils.ts:52]: Not sure from here! But I couldn't find a better place to trigger this event.

File: src/App.tsx [hunk 7]
```diff
@@ -29,6 +33,15 @@ const App = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  ipcRenderer.on('missing-dependency-result', (_, {dependencies}) => {
+    const alert: AlertType = {
+      type: AlertEnum.Warning,
+      title: 'Missing dependecy',
```
Comment (olensmar) [inline src/App.tsx:39]: typo - 'dependecy' -> 'dependency'

File: src/redux/thunks/utils.ts (hunk 6 above)
Comment (olensmar) [reply src/redux/thunks/utils.ts:52]: I think it's enough for now to just check for these dependencies on startup - we can remove this check for now

**Review by olensmar — APPROVED** [review]

**Pull request merged**